src/openms/source/FORMAT/PercolatorInfile.cpp

```diff
@@ -76,6 +80,111 @@ namespace OpenMS
+  {
+    CsvFile csv(pin_file, '\t');
+    StringList header;
+    csv.getRow(0, header);
+
+    unordered_map<String, size_t> to_idx; // map column name to column index
+    {
+      size_t idx{};
+      for (const auto& h : header) { to_idx[h] = idx++; }
+    }
+
+    // charge columns are not standardized so we check for the format and create hash to lookup column name to charge mapping
+    std::regex charge_one_hot_pattern("^charge\\d+$");
+    std::regex sage_one_hot_pattern("^z=\\d+$");
+    String charge_prefix;
+    unordered_map<String, int> col_name_to_charge;
+    for (const String& c : header)
+    {
+      if (std::regex_match(c, charge_one_hot_pattern))
+      {
+        col_name_to_charge[c] = c.substr(6).toInt();
+        charge_prefix = "charge";
+      }
+      else if (std::regex_match(c, sage_one_hot_pattern))
+      {
+        col_name_to_charge[c] = c.substr(2).toInt();
+        charge_prefix = "z=";
+      }
+      else if (c == "z=other") // SAGE
+      {
+        col_name_to_charge[c] = 0;
+      }
+    }
+
+    auto n_rows = csv.rowCount();
+    
+    vector<PeptideIdentification> pids;
+    pids.reserve(n_rows);
+    String spec_id;
+    for (size_t i = 1; i != n_rows; ++i)
+    {
+      StringList row;      
+      csv.getRow(i, row);
+
+      if (row.size() != header.size())
+      {
+        throw Exception::ParseError(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "Error: line " + String(i) + " of file '" + pin_file + "' does not have the same number of columns as the header!", String(i));
+      }
+
+      const String& sSpecId = row[to_idx.at("SpecId")];
+      if (sSpecId != spec_id)
+      {
+        pids.resize(pids.size() + 1);
+        pids.back().setHigherScoreBetter(higher_score_better);
+        pids.back().setScoreType(score_name);
+      }
+
+      int sScanNr = row[to_idx.at("ScanNr")].toInt();
+
+      String sPeptide = row[to_idx.at("Peptide")];
+      const double score = row[to_idx.at(score_name)].toDouble();
+      String target_decoy = row[to_idx.at("Label")].toInt() == 1 ? "target" : "decoy";
+      const String& sProteins = row[to_idx.at("Proteins")];
+      int rank = to_idx.count("rank") ? row[to_idx.at("rank")].toInt() : 1;
+      StringList accessions;
+
+      int charge = 0;
+      for (const auto&[name, z] : col_name_to_charge)
+      {
+        if (row[to_idx.at(name)] == "1")
+        {
+          charge = z;
+          break;
+        }
+      }
+
+      sProteins.split(';', accessions);
+
+      // deduce decoy state from accessions if decoy_prefix is set
+      if (!decoy_prefix.empty())
+      {
+        target_decoy = std::all_of(accessions.begin(), accessions.end(), [&decoy_prefix](const String& acc) { return acc.hasPrefix(decoy_prefix); }) ? "decoy" : "target" ;
+      }          
+
+      // needs to handle strings like: [+42]-MVLVQDLLHPTAASEAR, [+304.207]-ETC[+57.0215]RQLGLGTNIYNAER
+      sPeptide.substitute("]-", "]."); // we can parse [+42].MVLVQDLLHPTAASEAR
```

Comment:
Hmm interesting. What does `.[+42]` represent then?

Reply:
This should be N-term Acetyl. If the first match is not what we want we might need to figure out a way to prioritize modification resolution.

Reply:
No, I mean, of we are parsing [+42]. as nterm, what is .[+42]

I honestly thought we parse it the latter way.

Reply:
we support both afaik

Reply:
Ok then it would be fine. Did this happen in the test?

Reply:
yeah I check for this in the test because I wasn't sure